index.d.ts

```diff
@@ -1,212 +1,145 @@
-// Prior contributors: Torsten Werner <https://github.com/torstenwerner>
-//                     Niklas Lindgren <https://github.com/nikcorg>
-//                     Vinay Bedre <https://github.com/vinaybedre>
-//                     Antonio Román <https://github.com/kyranet>
-//                     Andrew Leedham <https://github.com/AndrewLeedham>
-//                     Jason Li <https://github.com/JasonLi914>
-//                     Brandon Wilson <https://github.com/wilsonianb>
-//                     Steve Faulkner <https://github.com/southpolesteve>
-
 /// <reference types="node" />
 
-import {URL, URLSearchParams} from 'url';
-import {Agent} from 'http';
-
-export class Request extends Body {
-	method: string;
-	redirect: RequestRedirect;
-	referrer: string;
-	url: string;
-
-	// Node-fetch extensions to the whatwg/fetch spec
-	agent?: Agent | ((parsedUrl: URL) => Agent);
-	compress: boolean;
-	counter: number;
-	follow: number;
-	hostname: string;
-	port?: number;
-	protocol: string;
-	size: number;
-	timeout: number;
-	highWaterMark?: number;
-
-	context: RequestContext;
-	headers: Headers;
-	constructor(input: string | { href: string } | Request, init?: RequestInit);
-	static redirect(url: string, status?: number): Response;
-	clone(): Request;
-}
-
-export interface RequestInit {
-	// Whatwg/fetch standard options
-	body?: BodyInit;
-	headers?: HeadersInit;
-	method?: string;
-	redirect?: RequestRedirect;
-	signal?: AbortSignal | null;
-
-	// Node-fetch extensions
-	agent?: Agent | ((parsedUrl: URL) => Agent); // =null http.Agent instance, allows custom proxy, certificate etc.
-	compress?: boolean; // =true support gzip/deflate content encoding. false to disable
-	follow?: number; // =20 maximum redirect count. 0 to not follow redirect
-	size?: number; // =0 maximum response body size in bytes. 0 to disable
-	timeout?: number; // =0 req/res timeout in ms, it resets on redirect. 0 to disable (OS limit applies)
-	highWaterMark?: number; // =16384 the maximum number of bytes to store in the internal buffer before ceasing to read from the underlying resource.
+import { Agent } from 'http';
+import { URL, URLSearchParams } from 'url'
+import { AbortSignal } from 'abort-controller'
```

Comment:
`abort-controller` is not a dependency so it won't work out of the box. Also, node-fetch is supposed to be a lightweight implementation and adding this might be weighing it down.

Reply:
@Richienb actually `abort-controller` is a `devDependency` of this project for quite a long time. What will be your proposal alternative to referring `AbortSignal` here - bring it from `lib.dom` or copy the whole definition inside (it's quite big)?

Reply:
Is it possible to import from `lib.dom` using ES6 imports?

Reply:
Nope, `lib.dom` is a library, not a module. And you probably should double-check the @jstewmon comment above about downside of just referring `lib.dom` in a triple-slash comment.